contracts/lens/FeeOnTransfer.sol

```diff
@@ -0,0 +1,117 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity =0.7.6;
+pragma abicoder v2;
+
+import '@uniswap/v3-periphery/contracts/base/PeripheryImmutableState.sol';
+import '@uniswap/v2-core/contracts/interfaces/IUniswapV2Callee.sol';
+import '@uniswap/v2-core/contracts/interfaces/IUniswapV2Pair.sol';
+import '../libraries/UniswapV2Library.sol';
+import '../interfaces/ISwapRouter02.sol';
+import '../interfaces/IFeeOnTransfer.sol';
+import '../interfaces/IERC20.sol';
+import '../interfaces/IWETH.sol';
+
+/// @notice Determine if a token takes a fee on transfer by flash borrowing from the token/<base token> pool on V2.
+/// @notice Returns true if we detected a fee is taken on transfer.
+/// @notice A return value of false does not mean the token is definitely not a fee on transfer token. It means
+/// @notice we are *unsure* if the token will take a fee on transfer.
+/// @dev We can not guarantee the result of this lens is correct for a few reasons:
+/// @dev 1/ Some tokens take fees under specific conditions, for example some have an allowlist
+/// @dev    of addresses that do/dont require fees. Therefore the result is not guaranteed to be correct
+/// @dev    in all circumstances.
+/// @dev 2/ It is possible that the token does not have any pools on V2 therefore we are not able to perform
+/// @dev    a flashloan to test the transfer.
+contract FeeOnTransfer is IFeeOnTransfer, IUniswapV2Callee, PeripheryImmutableState {
+    string internal constant FOT_REVERT_STRING = 'FOT';
+    bytes4 internal constant PAIR_TOKEN_0_SELECTOR = bytes4(keccak256('token0()'));
```

Comment:
should use the selector from the interface here: https://unpkg.com/@uniswap/v2-core@1.0.1/contracts/interfaces/IUniswapV2Pair.sol